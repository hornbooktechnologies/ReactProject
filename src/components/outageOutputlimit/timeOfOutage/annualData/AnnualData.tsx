import React, { Fragment } from 'react'
import AnnualImpact from './AnnualImpact'
import { useTranslation } from 'react-i18next'
import { TimeOfOutageModel } from '../../../../models/outageOutputLimit/TimeOfOutageModel'
import CustomPie from '../../../common/customPie/CustomPie'

const AnnualData = ({
  donutchartAnnualData,
  annualImpact,
}: TimeOfOutageModel) => {
  const { i18n } = useTranslation();
  const pieChartRenderSize = 240;
  return (
    <>
      <div className="uk-width-1-3@m media-mb-20">
        <div className="uk-card uk-height-1-1">
          <h5 className="mb-24 font-18">{i18n.t('LABELS.ANNUAL')}</h5>
          <div
            className="mb-40 ml--25 uk-grid uk-child-width-1-2@s uk-child-width-1-2@m"
            uk-grid="true"
          >
            {annualImpact?.map(
              (impactObj: { impactVal: number , impactPreviousVal :number, suffix:string,variation:string }, index: number) => {
                return (
                  <Fragment key={index}>
                    <AnnualImpact
                      impactVal={impactObj.impactVal} 
                      impactPreviousVal={impactObj.impactPreviousVal} 
                      suffix={impactObj.suffix} 
                      variation={impactObj.variation}/>
                  </Fragment>
                )
              }
            )}
          </div>
          {donutchartAnnualData !== null &&
            donutchartAnnualData !== undefined && (
            <CustomPie 
              data={donutchartAnnualData} 
              width={pieChartRenderSize}
              height={pieChartRenderSize}
              />
            )}
        </div>
      </div>
    </>
  )
}

export default AnnualData
