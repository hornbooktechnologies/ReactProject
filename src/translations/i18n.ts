import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
