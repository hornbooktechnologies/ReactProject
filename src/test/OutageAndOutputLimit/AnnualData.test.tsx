import React from "react";
import renderer from "react-test-renderer";
import AnnualData from "../../components/outageOutputlimit/timeOfOutage/annualData/AnnualData";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";

describe("Component: AnnualData", () => {
  it("AnnualData renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <AnnualData/>
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
