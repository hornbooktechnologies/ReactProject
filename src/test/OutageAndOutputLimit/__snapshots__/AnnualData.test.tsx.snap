// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AnnualData AnnualData renders correctly 1`] = `
<div
  className="uk-width-1-3@m media-mb-20"
>
  <div
    className="uk-card uk-height-1-1"
  >
    <h5
      className="mb-24 font-18"
    >
      LABELS.ANNUAL
    </h5>
    <div
      className="mb-40 ml--25 uk-grid uk-child-width-1-2@s uk-child-width-1-2@m"
      uk-grid="true"
    />
  </div>
</div>
`;
